packages/service-clients/end-to-end-tests/azure-client/src/test/viewContainerVersion.spec.ts

```diff
@@ -4,168 +4,242 @@
  */
 
 import { strict as assert } from "node:assert";
+import { AxiosResponse } from "axios";
 
 import { AzureClient } from "@fluidframework/azure-client";
 import { ConnectionState } from "@fluidframework/container-loader";
-import { ContainerSchema } from "@fluidframework/fluid-static";
+import { ContainerSchema, type IFluidContainer } from "@fluidframework/fluid-static";
 import { SharedMap } from "@fluidframework/map/internal";
 import { timeoutPromise } from "@fluidframework/test-utils/internal";
 
-import { createAzureClient } from "./AzureClientFactory.js";
-
-describe("viewContainerVersion scenarios", () => {
-	const connectTimeoutMs = 10_000;
-	let client: AzureClient;
-	const schema = {
-		initialObjects: {
-			map1: SharedMap,
+import {
+	createAzureClient,
+	createContainerFromPayload,
+	getContainerIdFromPayloadResponse,
+} from "./AzureClientFactory.js";
+import * as ephemeralSummaryTrees from "./ephemeralSummaryTrees.js";
+
+const testMatrix = [
+	{
+		variant: "Durable Container",
+		options: {
+			isEphemeral: false,
 		},
-	} satisfies ContainerSchema;
-
-	beforeEach("createAzureClient", () => {
-		client = createAzureClient();
-	});
-
-	beforeEach("skipForNonAzure", async function () {
-		if (process.env.FLUID_CLIENT !== "azure") {
-			this.skip();
-		}
-	});
+	},
+	{
+		variant: "Ephemeral Container",
```

Comment:
Nothing necessarily wrong with testing this, though I'm not exactly sure what the guarantees are on the ability to load old versions of an ephemeral container.  Probably not a real user scenario to both want a container to be ephemeral but also able to restore old versions of it?